src/workers/event.rs

```diff
@@ -0,0 +1,545 @@
+    }
+
+    // END HANDLER FUNCTIONS
+    async fn populate_actor_info(
+        &self,
+        mut actor_ids: HashSet<String>,
+        host_id: &str,
+        count_map: &HashMap<String, usize>,
+    ) -> anyhow::Result<Vec<(String, Actor)>> {
+        let resp = self
+            .ctl_client
+            .get_claims()
+            .await
+            .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+        Ok(resp
+            .claims
+            .into_iter()
+            .filter_map(|mut claim| {
+                let sub = claim.get("sub").map(|s| s.as_str()).unwrap_or_default();
+
+                actor_ids.take(sub).map(|id| {
+                    (
+                        id.clone(),
+                        Actor {
+                            id: id.clone(),
+                            name: claim.remove("name").unwrap_or_default(),
+                            capabilities: claim
+                                .remove("caps")
+                                .map(|raw| raw.split(',').map(|s| s.to_owned()).collect())
+                                .unwrap_or_default(),
+                            issuer: claim.remove("iss").unwrap_or_default(),
+                            count: [(
+                                host_id.to_owned(),
+                                count_map.get(&id).copied().unwrap_or_default(),
+                            )]
+                            .into(),
+                            ..Default::default()
+                        },
+                    )
+                })
+            })
+            .collect())
+    }
+
+    #[instrument(level = "debug", skip(self, host), fields(host_id = %host.id))]
+    async fn heartbeat_actor_update(
+        &self,
+        lattice_id: &str,
+        host: &HostHeartbeat,
+    ) -> anyhow::Result<()> {
+        debug!("Fetching current actor state");
+        let mut actors = self.store.list::<Actor>(lattice_id).await?;
+
+        let missing_actors: HashSet<String> = host
+            .actors
+            .iter()
+            .filter_map(|(id, _)| (!actors.contains_key(id)).then_some(id))
+            .cloned()
+            .collect();
+
+        // TODO: FIXME
+
+        let actors_to_update = host.actors.iter().filter_map(|(id, count)| {
+            match actors.remove(id) {
+                Some(mut current) => {
+                    // An actor count should never be 0, so defaulting is fine. If we didn't modify the count, skip
+                    if current
+                        .count
+                        .insert(host.id.clone(), *count)
+                        .unwrap_or_default()
+                        == *count
+                    {
+                        None
+                    } else {
+                        Some((id.to_owned(), current))
+                    }
+                }
+                None => None,
+            }
+        });
+
+        let actors_to_update: Vec<(String, Actor)> = if !missing_actors.is_empty() {
```

Comment:
So `missing_actors` covers the case where the lattice state we have doesn't include some actors on the heartbeat. But what about the reverse case, where the lattice state includes actors that aren't on the heartbeat? E.g. if we missed an ActorStopped event

Reply:
I addressed this with some stuff I added to the reaper process